src/FileWriterTask.h

```diff
@@ -25,48 +23,90 @@ class ParseError : public std::runtime_error {
-                                   std::string hdf_filename);
-  /// Used by Streamer to get the list of demuxers
+
+  /// Initialise the HDF file.
+  ///
+  /// \param NexusStructure The structure of the NeXus file.
+  /// \param ConfigFile The configuration information.
+  /// \param HdfInfo The HDF information for the stream.
+  /// \param UseSwmr Whether to use SWMR.
+  void InitialiseHdf(std::string const &NexusStructure,
+                     std::string const &ConfigFile,
+                     std::vector<StreamHDFInfo> &HdfInfo, bool UseSwmr);
+
+  /// \brief  Set the `JobID`.
+  ///
+  /// \param Id The Id value to use.
+  void setJobId(const std::string &Id);
+
+  /// \brief Add a source to the demuxers.
+  ///
+  /// \param source The source to add.
+  void addSource(Source &&Source);
+
+  /// \brief Set the filename.
+  ///
+  /// \param Prefix The path prefix.
+  /// \param Name The filename (can include path).
+  void setFilename(std::string const &Prefix, std::string const &Name);
+
+  /// \brief Get the list of demuxers.
+  ///
+  /// \return The demux topics.
   std::vector<DemuxTopic> &demuxers();
+
+  /// \brief  Get the unique numeric identifier of this job.
+  ///
+  /// Could maybe be replaced by `JobID`.
+  ///
+  /// \return The unique identifier.
   uint64_t id() const;
-  std::string job_id() const;
+
+  /// \brief  Get the job ID of the file being written.
+  ///
+  /// \return The job ID.
+  std::string jobID() const;
+
+  /// \brief  Return statistics about this job as JSON.
   nlohmann::json stats() const;
-  std::string hdf_output_prefix;
 
-  /// \brief  Name of the file being written
+  /// \brief  Name of the file being written.
+  ///
+  /// Important for reopening of files.
   ///
-  /// Important for reopen of files.
-  std::string hdf_filename;
-  std::string filename_full;
+  /// \return The file name.
+  std::string filename() const;
 
-  /// \brief  The file being written.
-  HDFFile hdf_file;
+  /// Get the group for the HDF file.
+  ///
+  /// \return The group.
+  hdf5::node::Group hdfGroup();
 
-  /// \brief  Whether we use HDF SWMR, initialized to the default.
-  bool UseHDFSWMR = true;
+  /// Get whether SWMR is enabled for this task.
+  ///
+  /// \return true if enabled.
+  bool swmrEnabled() const;
 
 private:
-  std::vector<DemuxTopic> _demuxers;
-  void add_source(Source &&source);
-  /// Called by CommandHandler on setup.
-  void hdf_init(std::string const &NexusStructure,
-                std::string const &ConfigFile,
-                std::vector<StreamHDFInfo> &stream_hdf_info);
-  void hdf_close_before_reopen();
-  int hdf_reopen();
-  void job_id_init(const std::string &);
-  uint64_t _id;
-  std::string JobID;
-  std::string ServiceID;
+  std::string Filename;
+  std::vector<DemuxTopic> Demuxers;
+  void closeFile();
+  void reopenFile();
+  uint64_t Id;
+  std::string JobId;
+  std::string ServiceId;
   std::shared_ptr<KafkaW::ProducerTopic> StatusProducer;
+  HDFFile hdfFile;
```

Comment:
Should be HdfFile

Reply:
@rerpha I disagree slightly, to prevent using the same (or almost the same name) for the type and the instance of that type, I suggest that the variable is instead named (e.g.) `File`.
```suggestion
  HDFFile File;
```